api/src/client/dynamics/DynamicsLicenseApplicationIdClient.ts

```diff
@@ -0,0 +1,109 @@
+import { LogWriterType } from "@libs/logWriter";
+import { LicenseStatus } from "@shared/license";
+import axios, { AxiosError } from "axios";
+import { LicenseApplicationIdClient, LicenseApplicationIdResponse } from "./types";
+
+export const DynamicsLicenseApplicationIdClient = (
+  logWriter: LogWriterType,
+  orgUrl: string
+): LicenseApplicationIdClient => {
+  const getLicenseApplicationId = (
+    accessToken: string,
+    businessId: string,
+    licenseType: string
+  ): Promise<LicenseApplicationIdResponse> => {
+    const logId = logWriter.GetId();
+    logWriter.LogInfo(`Dynamics License Application Id Client - Id:${logId}`);
+
+    return axios
+      .get(
+        `${orgUrl}/api/data/v9.2/rgb_applications?$select=rgb_appnumber,rgb_number,rgb_startdate,rgb_versioncode,rgb_expirationdate,statecode,statuscode,rgb_applicationid&$filter=(_rgb_businessid_value eq ${businessId} and _rgb_apptypeid_value eq ${licenseTypeToLicenseId[licenseType]})&$top=50`,
+        {
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+          },
+        }
+      )
+      .then((response) => {
+        logWriter.LogInfo(
+          `Dynamics License Application Id Client - Id:${logId} - Response: ${JSON.stringify(response.data)}`
+        );
+
+        const mainActiveApplications = response.data.value
+          .filter((applicationDetails: LicenseApplicationIdApiResponse) => applicationDetails.statecode === 0)
+          .filter(
+            (applicationDetails: LicenseApplicationIdApiResponse) =>
+              applicationDetails.rgb_number.slice(-2) === "00"
```

Comment:
[sand] The magic string `"00"` could be stored as a variable with a descriptive name. That would help with clarity

Reply:
Thinking about it now, even the `0` for statecode is a magic number and could be stored as an `ACTIVE_STATECODE` variable or something